### ISSUE TITLE
Separate Post Tags section

### ISSUE THREAD

**Issue by Justmadethisforforums** [bluesky-social/social-app#6216]
### Describe the Feature

Similar to the way Tumblr allows users to tag posts, separate from the post's main text and character limit.

Currently, the low character limit discourages people from using tags to label things like content warnings or relevant information [like #art, #illustration, etc] because they cut into the post's character limit. It also can be a bit ugly to have a whole string of hashtags, but it's necessary in today's world if you want to get noticed.

An alternative would be to have a [still character limited] 'tag section' that would go beneath the post, allowing people to still mark content with relative hastags that can be muted or browsed, but it doesn't cut into the main post or look as unsightly. 
This would greatly expand the capabilities of people to self-moderate, self-tag for content warnings, etc. 

To avoid people abusing the tags to make the post longer, just implement another character limit in the tagging section. If you can't fit all your tags in 240 characters, that's too many tags.

Here's how posts currently look:
![image](https://github.com/user-attachments/assets/3a7b3659-948d-4cd8-969f-16c55f63efdf)

Here's how they COULD look:
![image](https://github.com/user-attachments/assets/d073153c-1e44-49e9-9fdd-d468d94d5451)

Art post example:
![image](https://github.com/user-attachments/assets/7ed4d355-8093-41d6-830c-23f0538ec409)

### Describe Alternatives

An alternative would be that Hashtags don't count for your character limit, but that's way more easily exploitable and doesn't change the fact that a bunch of hashtags on the main post still looks kinda gross.

**Comment by mary-ext** [comment]
We do have outlined tags, it does exactly what you describe, but the use of hashtags has been rather mixed for curation that it hasn't exactly been a focus.

**Comment by Justmadethisforforums** [comment]
> We do have outlined tags, it does exactly what's described here. It's been implemented but bsky.app just doesn't have any UIs for it. Mainly because the use of hashtags for curation has been rather mixed so it's not been a focus.

Wait- it does??? Could you direct me where to find it? :0

**Comment by Justmadethisforforums** [comment]
> I like this as a way of separating metadata from prose content, a bunch of hashtags on the main post does look gross. If implemented it would be nice to add a user preference to hide hashtags on posts if you don't care about them.

Exactly- the tags would still apply to your filters/moderation, but you don't have to see them if you don't want. [Could also put a 1-2 line visual limit so it doesn't clog things up too much if you have them turned on]

**Comment by Tamschi** [comment]
> We do have outlined tags, it does exactly what's described here. It's been implemented but bsky.app just doesn't have any UIs for it. Mainly because the use of hashtags for curation has been rather mixed so it's not been a focus.

That may be a chicken-and-egg problem. Many people avoid too many inline hashtags because they feel cluttered and mess with accessibility, but outlined tags generally have much higher uptake.

I think a good way to drastically increase uptake would be to have the UI present some of the user's most frequent recent (hash)tags to be added with one click/tap, and if that doesn't fill the space then the most frequent among the last ~100 they've seen while scrolling. I think it's fine to go by UI space rather than absolute count for something like this, but there also needs to be a way to indefinitely exclude certain tags from the suggestions overall.

Especially for e.g. gallery curation, being able to edit posts to add these retroactively would be very important, though, so I'm pretty sure that if it's a choice between this and editing then the latter would be more welcomed.

> > We do have outlined tags, it does exactly what's described here. It's been implemented but bsky.app just doesn't have any UIs for it. Mainly because the use of hashtags for curation has been rather mixed so it's not been a focus.
> 
> Wait- it does??? Could you direct me where to find it? :0

Certain alternative clients like [deck.blue](https://bsky.app/profile/deck.blue) support creating and displaying them.

Even if a post has outlined tags, they won't be visible on bsky.app though, even though they _do_ make a post show up in hashtag searches and direct tag searches.

**Comment by hikatamika** [comment]
Adding this feature to the app UI sooner better than later would do a lot for artists.

A lot of them avoid hashtags for aesthetic reasons, or due to being used to other social apps where too many tags in the caption penalizes their posts algorithmically. _Causing them to smuggle their hashtags in the alt text instead,_ which backfires on accessibility. Outlined tags are in the post schema, 3rd party apps are making use of them being there, would love to see it first party as well.

**Comment by cooljeanius** [comment]
Note that Mastodon has this feature, too, for an additional point of comparison

**Comment by 9ineBand** [comment]
I think it would be a win for everyone. Even people who don't use it, may start using it just because it's there.

**Comment by Tamschi** [comment]
For what it's worth, if you use them in deck.blue, they'll affect hashtag searches in the main frontend for everyone else already.
You'll still need to talk to other people to get visibility, though. Bluesky is quite different from other large social media in that regard.
